hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -691,7 +689,62 @@ public InternalPartition getPartition(int partitionId, boolean triggerOwnerAssig
 
     @Override
     public boolean prepareToSafeShutdown(long timeout, TimeUnit unit) {
-        return partitionReplicaStateChecker.prepareToSafeShutdown(timeout, unit);
+        if (!node.joined()) {
+            return true;
+        }
+
+        if (node.isLiteMember()) {
+            return true;
```

Comment:
How do we handle the ongoing tasks run by executor service? Is it in scope of "fault tolerant executor service"?